### PR TITLE
Documentation to add new resource support using ACK

### PR DESCRIPTION
**Pull request by muvaf**
Signed-off-by: Muvaffak Onus <onus.muvaffak@gmail.com>

### Description of your changes

I was planning to write this after I get referencer field generation as well but several community members are asking how they can contribute to provider-aws using ACK so I decided to put it up there and make changes as we go.

Fixes https://github.com/crossplane/provider-aws/issues/482

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Not needed.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: CODE_GENERATION.md
Comment (muvaf) [inline CODE_GENERATION.md:277]: I will make it ready to merge after this PR is merged.

File: CODE_GENERATION.md
Comment (hasheddan) [inline CODE_GENERATION.md:12]: might be worth linking the URL of ACK repository here

File: CODE_GENERATION.md
Comment (hasheddan) [inline CODE_GENERATION.md:99]:
```suggestion
since when a reference is given, their value is resolved after the creation. In
```

File: CODE_GENERATION.md
Comment (hasheddan) [inline CODE_GENERATION.md:101]:
```suggestion
alongside with its referencer and selector fields. To ignore, you need to add the
```

File: CODE_GENERATION.md
Comment (hasheddan) [inline CODE_GENERATION.md:127]: nipick:
```suggestion
to the identifier field of the resource in AWS. We try to stick with name if available
```

File: CODE_GENERATION.md
Comment (hasheddan) [inline CODE_GENERATION.md:229]:
```suggestion
Late initialization means that if a user doesn't have a desired value for a field
```

**Review by hasheddan — APPROVED** [review]
Thanks for this @muvaf! Added a few non-blocking comments, feel free to merge when you are ready :)

File: CODE_GENERATION.md
Comment (negz) [inline CODE_GENERATION.md:1]: Nit: I bet 90% of folks reading this will know already, but it may be worth using 1-2 sentences to explain what ACK is and how it relates to Crossplane (e.g. that they share a codegen pipeline but the actual Amazon controllers are a separate thing).

File: CODE_GENERATION.md
Comment (negz) [inline CODE_GENERATION.md:32]: Is this a `kind` name (e.g. `DBInstance`) or the fully qualified name (e.g. `dbinstances.rds.aws.crossplane.io`)?

File: CODE_GENERATION.md
Comment (negz) [inline CODE_GENERATION.md:39]: Should we open issues against ACK, or Crossplane?

File: CODE_GENERATION.md
Comment (negz) [inline CODE_GENERATION.md:51]: Does `Stage` have special meaning to ACK, or is this just an example kind? I thought the former at first, but now I suspect it's the latter. I think it's worth clarifying in the document.

File: CODE_GENERATION.md
Comment (negz) [inline CODE_GENERATION.md:290]: What are our unit testing practices for ACK-generated controllers? It makes sense to skip unit tests for generated code, but should we be writing tests for the custom parts that are written by humans?

File: CODE_GENERATION.md
Comment (hiteshghia-splunk) [reply CODE_GENERATION.md:32]: I believe  it is the `kind` name - DBInstance

File: CODE_GENERATION.md
Comment (hiteshghia-splunk) [inline CODE_GENERATION.md:56]: Some explanation of what  this command is suppose to do will be helpful. for eg creates the crd.yaml.

Comment (hiteshghia-splunk) [reply CODE_GENERATION.md:56]: Also do we manually manually need to run https://github.com/crossplane/provider-aws/blob/master/apis/generate.go?

File: CODE_GENERATION.md
Comment (muvaf) [reply CODE_GENERATION.md:39]: It should be ACK since the code is there. Added some clarification.

File: CODE_GENERATION.md
Comment (muvaf) [reply CODE_GENERATION.md:51]: It's just an example kind, added a sentence to make it clear.

File: CODE_GENERATION.md
Comment (muvaf) [reply CODE_GENERATION.md:290]: Just added a section there. I think we need to write tests for every custom code we put in but other than that, tests for generated code should be OK to skip.

File: CODE_GENERATION.md
Comment (muvaf) [reply CODE_GENERATION.md:56]: Just added some details.

**Comment by muvaf** [comment]
This is ready to go. @negz @hiteshghia-splunk let me know if you need another look, otherwise I'm planning to merge this in a few hours as @hasheddan already gave his thumbs up.

**Pull request merged**
